packages/react-dom-bindings/src/client/ReactDOMInput.js

```diff
@@ -276,8 +320,19 @@ export function initInput(
     node.defaultChecked = !!initialChecked;
   }
 
-  if (name !== '') {
+  // Name needs to be set at the end so that it applies atomically to connected radio buttons.
+  if (
+    name != null &&
+    typeof name !== 'function' &&
+    typeof name !== 'symbol' &&
+    typeof name !== 'boolean'
+  ) {
+    if (__DEV__) {
+      checkAttributeStringCoercion(name, 'name');
+    }
     node.name = name;
+  } else {
+    node.removeAttribute('name');
```

Comment:
is it already unset? or for hydration it's not? in which case do you still need the previous clearing that was in the old code?

Reply:
Yea, I missed this one. I'll remove. It's a tricky one because it's a hydration error in DEV regardless. We don't try to patch up other things so is there a reason we'd consider this case important? Seems like not.

In fact, I have another todo to reconsider if we should even call this method at all during hydration. Seems like we should dispatch a change event if it's different and then trigger the restore mechanism but that would have to happen after commit.